### DISCUSSION TITLE
Scene viewer don’t work with duplicit names

### DISCUSSION THREAD

**Discussion by pavelseps** [google/model-viewer#3641] [Q&A]
### Description
Hello,
I found problems with developing for scene viewer but I'm not sure if I can report it here.
I have exported glTF/glb file from javascript library threejs (custom scene and objects). Exported files are valid by [glTF-Validator](https://github.khronos.org/glTF-Validator/). I tested files in your own validator [(https://arvr.google.com/scene-viewer-preview).](https://arvr.google.com/scene-viewer-preview) Everything look fine and valid. Loaded files in html `<model-viewer>` works fine too. 
Problem started when I want open files for AR preview (href=„intent://arvr.google.com/scene-viewer/1.0?….“). Exported files don’t show properly. Often 3D model are merged together. After long investigation I figured out problem. Bug appears when I use duplicit names for 3D object. I have multiple 3D groups with objects inside which have same names as in another 3D groups. ARVR thought it was same objet. From my point of view is ok use same names in different 3D groups and validators approves it. It will be nice if it can work or show error in your validator. It was pain figured it without any errors or debug messages. For now I fixed it by adding random string after names.

If you want I can give you example of broken glTF files.

### Browser Affected
In website model-viewer woks fine.

### OS

- [x] Android 10
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

**Comment by elalish** [comment]
@tpsiaki Another scene-viewer glTF parsing bug for you.
